sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs

```diff
@@ -24,9 +23,24 @@ public class AuthenticationEventResponseHandler : IValueBinder
         /// <summary>The response property.</summary>
         internal const string EventResponseProperty = "$event$response";
 
+        private AuthenticationEventResponse _response;
+
         /// <summary>Gets or sets the action result.</summary>
         /// <value>The action result.</value>
-        public AuthenticationEventResponse Response { get; internal set; }
+        public AuthenticationEventResponse Response
+        {
+            get => _response;
+            internal set
+            {
+                if (value != null)
```

Comment:
We only want to set Response if the AuthenticationEventResponse is not null. Throwing here would introduce duplicate validation code.

We have validation errors to nulls on either `Response` or `Response.Actions`. You can find tests for it in PayloadTests.cs

Reply:
If I understand you correctly, `null` was already rejected by validation earlier in the call chain.  In that case, it is not a value that we expect to see here and, if we do, something unexpected happened.  Is that correct?

Reply:
Response validation happens towards the end. So if this is set to null, it would still throw response validation, but just towards the end. 

But yes, if somehow a user of the nuget is able to set the value for response to null, then that is unexpected and our response validation is lacking.

Reply:
Is it possible to not set `_response` at all?

Reply:
It is not possible to set `_response` directly but it is possible to set `Response` and if it was set, we have set of validation that run on the value of it to make sure it is not null and that `Response` object matches the contract we have in place with ESTS

Reply:
unresolved the comment while we are conversating on this. (easier for me to find)

Reply:
> It is not possible to set `_response` directly but it is possible to set `Response` and if it was set, we have set of validation that run on the value of it to make sure it is not null and that `Response` object matches the contract we have in place with ESTS

Yes, but is it possible to create an instance of `AuthenticationEventResponseHandler`, never set anything to the `Response` property and then get into validation that would fail because `_response` is null?

Reply:
Specifically, `Response` property is set in `SetValueAsync`. This method is called via `IValueBinder` interface in the WebJobs internals, so we have no guarantee that it will be ever called or what would be the value. `Response` getter is available publicly and can be used anywhere, while setter is internal and can only be set inside library itself. Hence, I don't understand why we need to swallow incorrect value instead of failing immediately?

Reply:
>This method is called via IValueBinder interface in the WebJobs internals, so we have no guarantee that it will be ever called or what would be the value.

Why is that the case? I debugged through some tests locally with null values and unset values for response and it throws validation error every time. So even though a value is not set on `Response`

Thanks for pointing this out though. I do not like about how validation is tied to the setting of the value. Is there a way for me tie validation (and possibly other function calls) to the timeline of the event trigger? To trigger validation before the response is returned?

Reply:
Because without this, I am unsure how to throw a validation error as anyone can initialize `AuthenticationEventResponseHandler` without setting response. Other option is to set the default constructor to private

Reply:
modified response to be private & AuthenticationEventResponseHandler to be internal. This will make sure that client is not able to initialize